st2common/st2common/services/config.py

```diff
@@ -0,0 +1,148 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Service functions for managing pack configuration inside the datastore.
+"""
+
+import json
+
+from st2common import log as logging
+from st2common.services import keyvalues as keyvalue_service
+from st2common.models.db.keyvalue import KeyValuePairDB
+from st2common.persistence.keyvalue import KeyValuePair
+
+__all__ = [
+    'get_datastore_key_name',
+
+    'get_datastore_value_for_config_key',
+    'set_datastore_value_for_config_key',
+
+    'get_datastore_value'
+]
+
+LOG = logging.getLogger(__name__)
+
+DATASTORE_KEY_SEPARATOR = '.'
+
+# Prefix for datastore items which store config values
+# Full keys follow this format: system.pack_config.<pack name>.<config key name>
+# For example:
+# pack_config.aws.setup.region
+DATASTORE_CONFIG_KEY_PREFIX = ['system', 'pack_config']
```

Comment:
Not a fan of pack_config. I actually want to use config.pack_name.foo instead of system.pack_config. This important UX decision needs the eyes of Sir @dzimine, Sir @manasdk.